### ISSUE TITLE
Style/IfUnlessModifier - Potential False Positive as of 0.46.0

### ISSUE THREAD

**Issue by calebtocco** [sds/haml-lint#512]
### Expected Behavior

No warning if using modifier results in line length warning on subsequent line.

### Observed Behavior

```
- if policy(record).update?
  = link_to 'Edit Record', edit_record_path(record), class: 'btn btn-secondary btn-sm btn-block' 
```

Warning:
`Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.`

### Rubocop Doc Example

```
if long_condition_that_prevents_code_fit_on_single_line
  do_something_with_a_long_name(arg)
end

```

**Comment by MaxLap** [comment]
Is this really a false positive?

This would work, no?
```
= link_to 'Edit Record', edit_record_path(record), class: 'btn btn-secondary btn-sm btn-block' if policy(record).update?
```

**Comment by MaxLap** [comment]
I understand the annoyance.

The problems:
* There are 2 linters interacting together: RuboCop and all the other linters of HamlLint. 
* The control we have over RuboCop is quite minimal. 
* The code we generate for RuboCop doesn't have the same length as the one in your HAML file. Ex: `= foo` becomes `HL.out = foo`.

I don't see how RuboCop could properly handle the line length without having tons of edge cases. Right now, RuboCop's LineLength is turned off because it would be mostly meaningless.

Maybe giving RuboCop a shorter LineLength rule than HamlLint's configured one would help. To avoid people getting bad lints from RuboCop's LineLength cop, we would need to ignore those (remove them).

Then, RuboCop would try to avoid going over this limit which is shorter than HamlLint, and so hopefully would fix your case.

I'm a bit worried this might end up being a bit of a Whac-A-Mole game. With some code getting split on multiple lines when it wasn't needed because RuboCop thought the line needed to be shorter than it really needed to.

I don't have the bandwidth to investigate that, but if a PR wants to try to make it work, I'll take a look.

The other option is not to use the LineLength Linter or disable the IfUnlessModifier Cop. Both of which can be done on the user side.
